components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/impl/CORSManagerImpl.java

```diff
@@ -0,0 +1,142 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.cors.service.internal.impl;
+
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.configuration.mgt.core.ConfigurationManager;
+import org.wso2.carbon.identity.configuration.mgt.core.exception.ConfigurationManagementException;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Resource;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.cors.mgt.core.constant.ErrorMessages;
+import org.wso2.carbon.identity.cors.mgt.core.exception.CORSManagementServiceClientException;
+import org.wso2.carbon.identity.cors.mgt.core.exception.CORSManagementServiceException;
+import org.wso2.carbon.identity.cors.mgt.core.exception.CORSManagementServiceServerException;
+import org.wso2.carbon.identity.cors.mgt.core.internal.function.ResourceToCORSConfiguration;
+import org.wso2.carbon.identity.cors.mgt.core.internal.function.ResourceToCORSOrigin;
+import org.wso2.carbon.identity.cors.mgt.core.model.CORSConfiguration;
+import org.wso2.carbon.identity.cors.mgt.core.model.CORSOrigin;
+import org.wso2.carbon.identity.cors.service.CORSManager;
+import org.wso2.carbon.identity.cors.service.internal.CORSServiceHolder;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.wso2.carbon.identity.cors.mgt.core.internal.Constants.CORS_CONFIGURATION_RESOURCE_NAME;
+import static org.wso2.carbon.identity.cors.mgt.core.internal.Constants.CORS_CONFIGURATION_RESOURCE_TYPE_NAME;
+import static org.wso2.carbon.identity.cors.mgt.core.internal.Constants.CORS_ORIGIN_RESOURCE_NAME;
+import static org.wso2.carbon.identity.cors.mgt.core.internal.Constants.CORS_ORIGIN_RESOURCE_TYPE_NAME;
+
+/**
+ * CORSManager implementation.
+ */
+public class CORSManagerImpl implements CORSManager {
+
+    private static final CORSManagerImpl instance = new CORSManagerImpl();
+
+    public CORSManagerImpl() {
+
+    }
+
+    public static CORSManagerImpl getInstance() {
+
+        return instance;
+    }
+
+    @Override
+    public List<CORSOrigin> getCORSOrigins(String tenantDomain) throws CORSManagementServiceClientException,
+            CORSManagementServiceServerException {
+
+        // Validate the tenant domain
+        validateTenantDomain(tenantDomain);
+
+        try {
+            FrameworkUtils.startTenantFlow(tenantDomain);
+
+            Resource resource = getConfigurationManager().getResource(CORS_ORIGIN_RESOURCE_TYPE_NAME,
+                    CORS_ORIGIN_RESOURCE_NAME);
+            List<CORSOrigin> corsOrigins;
+            if (resource == null) {
+                corsOrigins = new ArrayList<>();
```

Comment:
we can remove Collections unmodifiable empty list from here

Reply:
@senthalan I didn't get you. Can you please elaborate?